### ISSUE TITLE
More documentation on PETSc preconditioners

### ISSUE THREAD

**Issue by YaqiWang** [idaholab/moose#16516]
## Reason
We have a great documentation on one of the preconditioners (HYPRE BoomerAMG) here https://mooseframework.inl.gov/application_development/hypre.html. But we are missing the similar documentation on other preconditioning types, such as `lu`, `ilu`, `hmg`, etc.

## Design
It will be nice to have a list of PETSc options with some lay-man explanations for each pc type and catalog them based on their importance. I also found FSP is a little difficult to use but can be extremely useful. It will be great if we can document its usage somewhere as well. I do not think we need to have a full list of options because PETSc is changing although it is nice if we can. Some more other options related with linear/nonlinear solvers (https://mooseframework.inl.gov/source/executioners/Steady.html) can possibly be added as well. For example, the one for matrix debugging is not in the list in that page. Also some options affecting Jacobian-free algorithm are in the text but not in the table.

## Impact
Improved user interface and easier user support (we can direct users to these pages).

**Comment by WilkAndy** [comment]
Good idea.  Much of this should really be done by the PETSc people, but i find their documentation difficult to find and read.   I end up searching old mailing lists and so on for tips about what to use, and how to use them.  I wish we had a collective "brain" on these sorts of things, because i bet my models would run faster with optimal choices.
